### ISSUE TITLE
Add null checks to constructors

### ISSUE THREAD

**Issue by jzheaux** [spring-projects/spring-security#6892]
Related to #6542 

`RequestKey`, `JaasGrantedAuthority`, and `SwitchUserGrantedAuthority` each anticipate certain `final` members will be non-null without requiring such in the constructor.

For example, `RequestKey` does the following in `equals`:

```java
if (!url.equals(key.url)) {
    return false;
}
```

Where `url` is a member variable. But, the constructor does nothing to ensure this behavior will actually work:

```java
public RequestKey(String url, String method) {
    this.url = url;
    this.method = method;
}
```

For `RequestKey`, we should change the constructor to something like:

```java
public RequestKey(String url, String method) {
    Assert.notNull(url, "url cannot be null");
    this.url = url;
    this.method = method;
}
```

And we'd do similar things for the other two classes.

**Comment by clementkng** [comment]
Hi @jzheaux, is this issue open for contribution? If so, I would like to try working on this since I'm interested in learning more about this project.

**Comment by jzheaux** [comment]
Yes, @clementkng, it certainly is. Thanks for your interest! The ticket is yours.

**Comment by dfcoffin** [comment]
@jzheaux is there a list of which class constructors and values should be verified as containing non-null values?

**Comment by jzheaux** [comment]
Thanks for the question, @dfcoffin:

`RequestKey` -> `url`
`JaasGrantedAuthority` -> `role`, `principal`
`SwitchUserGrantedAuthority` -> `role`, `source`

**Comment by clementkng** [comment]
@jzheaux the integration test seems to be failing for me locally even though I am running `./gradlew clean build integrationTest` on master with no changes. I'm not sure if this OK behavior, and whether we're supposed to run the above command or `./gradlew build` suffices for testing.

**Comment by jzheaux** [comment]
What is the error you're seeing? Note that `integrationTest` on the full suite fails for JDK 11 on some of the samples.

**Comment by clementkng** [comment]
I'm currently running the project on the Windows Subsystem for Linux using JDK 10. Just wanted to make sure it's unexpected behavior before diving in. I'm getting a selenium.NoSuchElementException and a junit.ComparisonFailure.

![image](https://user-images.githubusercontent.com/24197092/58347816-b2c14600-7e13-11e9-8c80-58cd4a8dafe9.png)

**Comment by clementkng** [comment]
@jzheaux I've created a PR for the issue [here](https://github.com/spring-projects/spring-security/pull/6915)

**Comment by jzheaux** [comment]
Fixed via e66369f
